src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -222,10 +243,19 @@ private void BuildGlobbedLinkTags(IDictionary<string, string> attributes, TagHel
         private void BuildFallbackBlock(TagHelperContent builder)
         {
             EnsureGlobbingUrlBuilder();
-            var fallbackHrefs = GlobbingUrlBuilder.BuildUrlList(FallbackHref, FallbackHrefInclude, FallbackHrefExclude);
+            var fallbackHrefs =
+                GlobbingUrlBuilder.BuildUrlList(FallbackHref, FallbackHrefInclude, FallbackHrefExclude).ToArray();
 
             if (fallbackHrefs.Any())
             {
+                if (FileVersion == "true")
+                {
+                    for (var i=0; i < fallbackHrefs.Count(); i++)
+                    {
+                        fallbackHrefs[i] += "?v=" + GetFileContentEncodedString(fallbackHrefs[i]);
```

Comment:
Hmm we need to unify this logic, we shouldn't be doing this in two places.

Add a helper method called something like `string ResolveVersionedHref(string href)`. We can then use this in the `AppendHref` method that I left snippets in. In those snippets you can modify them to write out the "resolved" href value `if (FileVersion)`.

Ultimately we want to try and avoid strings as much as possible but for this specific case we can make due for the sake of unifying logic.

Once the helper method is completed this should be:

```C#
if (FileVersion)
{
    for (var i = fallbackHrefs.Count() - 1; i--; i >= 0)
    {
        fallbackHrefs[i] = ResolveVersionedHref(fallbackHrefs[i]);
    }
}
```